### ISSUE TITLE
Xcode code signing issue when distributing app with manual AWS setup

### ISSUE THREAD

**Issue by yoasha** [aws-amplify/aws-sdk-ios#2381]
**Describe the bug**
When using the latest version of the SDK (2.13.0), Xcode can't find a provisioning profile to sign my app's binary, I suspect it's because of AWS framework related issue.
With a previous version of AWS (2.12.3) there is no issue and Xcode completes the signing process successfully.
- I am using only two frameworks: AWSCore and AWSS3.
- I am using manual setup as described at https://aws-amplify.github.io/docs/ios/manualsetup#frameworks-setup (I'm not using Cocoa-pods)

Just to emphasize, I frequently built and distribute my app to the App Store without any issue while AWS frameworks are included. So my project is configured correctly with AWS SDK. The issue suddenly started after I switched from 2.12.3 to 2.13.0. 
If I rollback and use 2.12.3 instead of 2.13.0, then there is no code signing issue.

Any ideas?

![image](https://user-images.githubusercontent.com/5601548/77441718-6b6c6300-6df2-11ea-948e-303acae47a19.png)

**Comment by wooj2** [comment]
Hey @yoasha,

Thanks for reaching out!  Hmm.. there doesn't seem to be any difference in signing requirements between 2.12.x and 2.13.0. Have you tried to, clean you project, close XCode, Remove your derived files directory, and try rebuilding?

Also, is there a specific error you are getting when attempting to build with 2.13.0?  If so, can you please post what you are seeing?

Thanks again!

**Comment by yoasha** [comment]
Hi, 
Yes, I did clean the project, restarted Xcode, and even restarted my iMac. The issue is reproduced consistently when using 2.13.0 frameworks and disappears when using 2.12.3 frameworks. So it does look like a real issue.

Regarding the error, there is actually no error. Here is what happening after archiving my app and selecting "Distribute App" from the Organizer window:

Page #1 - Select a method of distribution (I select "App Store Connect")
Page #2 - Select a destination (I select "Upload")
Page #3 - Distribution options (I select both available options: "include bitcode" and "upload symbols")

With 2.12.3 frameworks:
page #4 - Re-sign "<MY APP NAME>" (I select "Automatically manage signing")
page #5 - The final review page where I can select the "Upload" button

With 2.13.0 frameworks:
page #4 - Select certificate and iOS App Store profiles. (see screenshot attached in my original report)
This is the unexpected page that never appeared for me in the past and also doesn't appear when using AWS 2.12.3 frameworks. I assume it pops up since Xcode can't find satisfying provisioning profile for my app. When I select the dropdown list to choose a profile, it shows a single provisioning profile which appears under "Ineligible" section, and if I select it, it will display various error indicators claiming that the selected provisioning profile is expired and also doesn't match the entitlements of my app. 

This page really confused me when I started to investigate this issue as I was certain that there is a signing issue related to my certificate or to my project settings. However, only after going deeper I found out that the appearance of this page is somehow related to the version of the AWS frameworks.

BTW, when I'm looking into any AWS framework package of 2.13.0 I can see _CodeSignature/CodeResources file. However, this file doesn't appear in the 2.12.3 AWS framework packages. I'm not sure if this is relevant but I though it worth mentioning as it seems to be related to code signing.

**Comment by wooj2** [comment]
1.  Hmm I doubt that the CodeResources file would have anything to do with not being able to submit your app, but you could always try deleting those files and see if that makes any difference.

2.  Have you deleted your derived files?

3.  I issued a google search, and not sure if this is the issue you're running into, but it seems like there is a reddit post with one follow up comment here:
https://www.reddit.com/r/iosdev/comments/ezuhsd/select_certificate_and_ios_app_store_profiles/

The post seems to indicate that maybe there is something within your xxcode project where you need to select automatically manage signing (within the xcode project)?  Have you tried this?

**Comment by yoasha** [comment]
Many thanks for your continues support.
1. I did try deleting these files but it didn't help. I just thought it might point to something related with code signing on your side..
2. Yes, I deleted all derived files, but it didn't make any difference.
3. It seems that the user posted the issue on reddit used manual code signing. I am using automatic code signing so this page should never appear theoretically. I also toggled the automatic code signing OFF and ON as suggested in some posts, but that didn't help.

My workspace has three different apps and this issue is reproduced on each project. I guess it's related to my unique configuration as I couldn't find any similar report on the internet. I am also not sure how many users are using the manual setup vs. Cocoapods setup, so maybe the issue is there but affecting small amount of users.

I will try to switch to Cocoapods setup and update here if it fixes my issue.

**Comment by yoasha** [comment]
UPDATE: I switched to Cocoapods setup and all issues are resolved.

**Comment by yemartin** [comment]
@yoasha We recently hit the same issue, and thanks to your report here, we downgraded to 2.12.3, and that fixed it. So thanks a lot! But while one may work around it by using Cocoapods, the problem remains:

**The manual AWS setup is currently broken**.

Hours were lost trying to diagnose the problem. And in our case, we do not want to use Cocoapods so we must now use an older version of the library...

I don't want to create a new issue that would just be a duplicate of this, so for the sake of the community at large, would you mind reopening this? Thank you.

**Comment by yoasha** [comment]
Reopening this issue per @yemartin request.

**Comment by yemartin** [comment]
Thank you for reopening @yoasha 

So yes, @wooj2 , the issue still exists in the latest release (2.14.0).
See below for a full bug report with just 6 steps to reproduce. Can you please take a look?

I tried 8 different versions, and it looks like the issue was introduced in 2.12.7:

2.12.3 ==> Automatic signing option, bitcode build success
2.12.4 ==> Automatic signing option, bitcode build success
2.12.5 ==> Automatic signing option, bitcode build success
2.12.6 ==> Automatic signing option, bitcode build success

2.12.7 ==> No automatic signing option, bitcode build error
2.12.8 ==> No automatic signing option, bitcode build error
2.13.0 ==> No automatic signing option, bitcode build error
2.14.0 ==> No automatic signing option, bitcode build error

### Prerequisite

* Latest Xcode (as of now: 11.5 (11E608c))
* Various SDK versions downloaded, in particular these three, to confirm that the issue was introduced in 2.12.7,
  and that it is still present in 2.14.0:
  https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.12.6.zip
  https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.12.7.zip
  https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.14.0.zip

### Steps to reproduce

1) *Create a new Xcode project*
  New > Project... > iOS > Single View App (Swift / SwiftUI, uncheck all checkboxes)
2) *Add AWSCore framework*
  Drag and drop `AWSCore.framework` in the project sidebar, check "Copy items if needed", "Create group" and
  add to targets. Click finish.
3) *Set Embed & Sign*
  In the target > General > Frameworks, Libraries, and Embedded Content, next to "AWSCore.framework",
  select "Embed & Sign"
4) *Add Strip Frameworks build phase*
  In Build Phases, add a "New Run Script Phase", call it "Strip Frameworks", and configure it as below:
```
  Shell /bin/sh
 
  bash "${BUILT_PRODUCTS_DIR}/${FRAMEWORKS_FOLDER_PATH}/AWSCore.framework/strip-frameworks.sh"
 
  Show environment variables in build log: Checked
  Run script only when installing: Not checked
  Use discovered dependency file: Not checked
 
  Input Files: Empty
  Output Files: Empty
```
5) *Archive the app*
  In the toolbar, select "Generic iOS Device", then Product > Archive
6) *Do an Ad Hoc distribution*
  In the organizer, select the built archive, and do Distribute > Ad Hoc > Next. Configure as follows:
```
App Thinning: None
Rebuild from Bitcode: Checked   <---- Important
Include manifest: Not checked
```
  and now click Next...

### Expected Behavior (with 2.12.3, 2.12.4, 2.12.5 and 2.12.6)

At this point, a "needs to be re-signed for Ad Hoc distribution" step in the wizard should show, reading:

> "your project name" needs to be re-signed for Ad Hoc distribution. Select one of the following signing options to continue:
> (*) Automatically manage signing
> ( ) Manually manage signing

After selecting "Automatically manage signing" and clicking next, Xcode contacts Apples servers, completes
the resign and bitcode recompile succesfully, and the last "Review IPA content" wizard page is shown.

### Actual Behavior (with 2.12.7, 2.12.8, 2.13.0 and 2.14.0)

We are not offered to let Xcode manage signing automatically. Instead, the wizard shows a page
"Select certificate and iOS Ad Hoc profiles", which is what we would see if we had chosen to
"Manually manage signing" above.

And even if we do configure the signing manually, the bitcode build will fail with:

```
ipatool failed with an exception: #<CmdSpec::NonZeroExitException: $ /Applications/Xcode.app/Contents/Developer/usr/bin/python3 /Applications/Xcode.app/Contents/Developer/usr/bin/bitcode-build-tool -v -t /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin --sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.5.sdk -o /var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/ipatool20200703-78931-17wc1ww/thinned-out/arm64/Payload/aws-amplify-framework-signing-error.app/aws-amplify-framework-signing-error --generate-dsym /var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/ipatool20200703-78931-17wc1ww/thinned-out/arm64/Payload/aws-amplify-framework-signing-error.app/aws-amplify-framework-signing-error.dSYM --strip-swift-symbols /var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/ipatool20200703-78931-17wc1ww/thinned-in/arm64/Payload/aws-amplify-framework-signing-error.app/aws-amplify-framework-signing-error
Status: pid 78963 exit 1
Stdout:
    SDK path: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.5.sdk
    SDK version: 13.5
    PATH: ['/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin', '/Applications/Xcode.app/Contents/Developer/usr/bin']
    Using: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo
    MachoInfo: cd /
     ['/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo', '-info', '/var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/ipatool20200703-78931-17wc1ww/thinned-in/arm64/Payload/aws-amplify-framework-signing-error.app/aws-amplify-framework-signing-error']
    -= Output =-
    Non-fat file: /var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/ipatool20200703-78931-17wc1ww/thinned-in/arm64/Payload/aws-amplify-framework-signing-error.app/aws-amplify-framework-signing-error is architecture: arm64
    Exited with 0
    
    Command took 0 seconds
    Using: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/dwarfdump
    GetUUID: cd /
     ['/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/dwarfdump', '-u', '/var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/ipatool20200703-78931-17wc1ww/thinned-in/arm64/Payload/aws-amplify-framework-signing-error.app/aws-amplify-framework-signing-error']
    -= Output =-
    UUID: F5C8999A-F42E-373E-9DF6-24CB6433E2DD (arm64) /var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/ipatool20200703-78931-17wc1ww/thinned-in/arm64/Payload/aws-amplify-framework-signing-error.app/aws-amplify-framework-signing-error
    Exited with 0
    
    Command took 0 seconds
    Using: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/segedit
    ExtractXAR: cd /
     ['/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/segedit', '/var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/ipatool20200703-78931-17wc1ww/thinned-in/arm64/Payload/aws-amplify-framework-signing-error.app/aws-amplify-framework-signing-error', '-extract', '__LLVM', '__bundle', '/var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/aws-amplify-framework-signing-error99hamx3b/aws-amplify-framework-signing-error.arm64.xar']
    -= Output =-
    Exited with 0
    
    Command took 0 seconds
    Bitcode bundle version: 1.0
    Setting platform to: iOS
    Using: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
    Using: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc
    Swift: cd /var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/templj7ykulb
     ['/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc', '-frontend', '-emit-object', '-target', 'arm64-apple-ios13.5', '-Xllvm', '-aarch64-use-tbi', '-O', '-disable-llvm-optzns', '-module-name', 'aws_amplify_framework_signing_error', '1.bc', '-o', '1.o']
    -= Output =-
    Exited with 0
    
    Command took 0 seconds
    Swift: cd /var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/templj7ykulb
     ['/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc', '-frontend', '-emit-object', '-target', 'arm64-apple-ios13.5', '-Xllvm', '-aarch64-use-tbi', '-O', '-disable-llvm-optzns', '-module-name', 'aws_amplify_framework_signing_error', '2.bc', '-o', '2.o']
    -= Output =-
    Exited with 0
    
    Command took 0 seconds
    Swift: cd /var/folders/bd/6d3xtd494zd6xw54b3sb1xd00000gn/T/templj7ykulb
     ['/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc', '-frontend', '-emit-object', '-target', 'arm64-apple-ios13.5', '-Xllvm', '-aarch64-use-tbi', '-O', '-disable-llvm-optzns', '-module-name', 'aws_amplify_framework_signing_error', '3.bc', '-o', '3.o']
    -= Output =-
    Exited with 0
    
    Command took 0 seconds
    Using: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
    Search Path: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.3/lib/darwin, /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphoneos, /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.5.sdk/usr/lib, /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.5.sdk/System/Library/Frameworks
    AWSCore not found in dylib search path

Stderr:

>
    /Applications/Xcode.app/Contents/Developer/usr/bin/ipatool:297:in `run'
    /Applications/Xcode.app/Contents/Developer/usr/bin/ipatool:2706:in `block in CompileOrStripBitcodeInBundle'
    /Applications/Xcode.app/Contents/Developer/usr/bin/ipatool:2645:in `each'
    /Applications/Xcode.app/Contents/Developer/usr/bin/ipatool:2645:in `CompileOrStripBitcodeInBundle'
    /Applications/Xcode.app/Contents/Developer/usr/bin/ipatool:2905:in `block in ProcessIPA'
    /Applications/Xcode.app/Contents/Developer/usr/bin/ipatool:2867:in `each'
    /Applications/Xcode.app/Contents/Developer/usr/bin/ipatool:2867:in `ProcessIPA'
    /Applications/Xcode.app/Contents/Developer/usr/bin/ipatool:3814:in `<main>'
```

**Comment by yemartin** [comment]
Hi @palpatim and thank you for looking into this. Let me know if there is anything I can do to help!

**Comment by palpatim** [comment]
It looks like the binary framework distributed in our Carthage archive (which is uploaded as an artifact to our GitHub release page) works as expected. @yemartin can you verify whether this works for you? We've been considering retiring the S3 distributed package since it's covered by the Carthage package, but if there's a use case we're missing, I'd like to know it.

To use:

- Download `aws-sdk-ios-carthage.framework.zip` files from one of the recent [releases](https://github.com/aws-amplify/aws-sdk-ios/releases)
- Unzip it locally
- The frameworks are located in the `Carthage/Build/iOS/` directory, along with the associated symbolication dSYMs and bcsymbolmaps
- Copy the AWSCore.framework framework to your test project, clean the Build folder & Derived Data, and repeat your test steps

Let me know if this works for you.

**Comment by yemartin** [comment]
Thank you @palpatim for the information. I can confirm that the framework from the Carthage package does fix the issue.

FYI, our use case: we are using Swift Package Manager, and do not want to multiply package managers within the same project for better developer experience. So until a Swift Package Manager build is availabe, I hope the direct framework installation remains an officially supported install method, but I am perfectly find with it piggy-backing on top of the Carthage build.

**Comment by palpatim** [comment]
@yemartin Thanks for the explanation. Maintaining the framework builds seems like a relatively low-friction way of serving customers like you who don't need YAPM (Yet Another Package Manager) in their lives :). I'll be pushing a PR for using Carthage builds as the basis for binary distribution.

**Comment by palpatim** [comment]
Pending https://github.com/aws-amplify/aws-sdk-ios/pull/2877

**Comment by palpatim** [comment]
Released on https://github.com/aws-amplify/aws-sdk-ios/releases/tag/2.15.1
